### ISSUE TITLE
"Remember password" doesn't work between multiple Linux installations

### ISSUE THREAD

**Issue by kparal** [ValveSoftware/steam-for-linux#3059]
I have 3 Steam installations - Fedora Desktop (#1), Windows Desktop (#2) and Fedora Notebook (#3). I have "Remember password" checked on every installation's login dialog.

I can easily do this without re-entering the password:
- Run steam on Fedora Desktop -> exit -> Run steam on Windows Desktop -> exit -> Run steam on Fedora Desktop
- Run steam on Fedora Notebook -> exit -> Run steam on Windows Desktop -> exit -> Run steam on Fedora Notebook

But I always get a password prompt when I try to do this:
- Run steam on Fedora Desktop -> exit -> Run steam on Fedora Notebook (password prompt forced on me here)
- Run steam on Fedora Notebook -> exit -> Run steam on Fedora Desktop (password prompt forced on me here)

I also tried Steam on Ubuntu and also on other Fedora installations in VirtualBox. Simply said, I can switch between different operating systems (even between Fedora and Ubuntu) without re-entering my password. But I **have to** enter my password every time I try to move from Fedora to a different Fedora.

This is very uncomfortable, because I usually switch between my desktop and notebook easily several times a day, and I simply can't enter my super-strong password every time I do that. I'll need to change it to something much weaker (and easier to type) very soon, just to stay sane.

There is nothing in the steam console output that would indicate why I'm required to re-type my password. I always correctly exit the Steam client on one system before running it on a different system. I also made sure that the "friendly name" (hostname) steam remembers is different on all systems.

To point out the important fact once again: This only happens when I switch between two Fedora systems (possibly also two Ubuntu systems, I haven't checked that), but not when I switch between two different operating systems.

Could somebody please find out what's going on and how to fix that?

```
Processor Information:
    Vendor:  GenuineIntel
    CPU Family:  0x6
    CPU Model:  0x3c
    CPU Stepping:  0x3
    CPU Type:  0x0
    Speed:  3600 Mhz
    4 logical processors
    4 physical processors
    HyperThreading:  Unsupported
    FCMOV:  Supported
    SSE2:  Supported
    SSE3:  Supported
    SSSE3:  Supported
    SSE4a:  Unsupported
    SSE41:  Supported
    SSE42:  Supported

Network Information:
    Network Speed:  

Operating System Version:
    Linux (64 bit)
    Kernel Name:  Linux
    Kernel Version:  3.12.5-302.fc20.x86_64
    X Server Vendor:  Fedora Project
    X Server Release:  11404000
    X Window Manager:  GNOME Shell
    Steam Runtime Version:  steam-runtime-release_2013-10-23

Video Card:
    Driver:  Intel Open Source Technology Center Mesa DRI Intel(R) Haswell Desktop 

    Driver Version:  3.0 Mesa 9.2.5
    OpenGL Version: 3.0
    Desktop Color Depth: 24 bits per pixel
    Monitor Refresh Rate: 59 Hz
    VendorID:  0x8086
    DeviceID:  0x412
    Number of Monitors:  1
    Number of Logical Video Cards:  1
    Primary Display Resolution:  1680 x 1050
    Desktop Resolution: 1680 x 1050
    Primary Display Size: 18.07" x 11.65"  (21.50" diag)
                                            45.9cm x 29.6cm  (54.6cm diag)
    Primary VRAM Not Detected

Sound card:
    Audio device: Intel Haswell HDMI

Memory:
    RAM:  7875 Mb

Miscellaneous:
    UI Language:  English
    LANG:  en_US.UTF-8
    Microphone:  Not set
    Total Hard Disk Space Available:  62641 Mb
    Largest Free Hard Disk Block:  44710 Mb

Installed software:

Recent Failure Reports:

```

**Comment by PureTryOut** [comment]
Sorry for bumping this, but this issue still exists, it's just not Fedora specific.

I use Arch Linux on both my desktop and laptop, and whenever I switch to either of them, I have to enter my password even though the "Remember my password" option is selected. I can switch to Windows and back without problems though.

**Comment by l3mde** [comment]
Still present for me under 2 Arch Linux Installations on laptop and desktop :(

**Comment by PureTryOut** [comment]
Yup still the same here too. If any more info is needed i'll be happy to provide it.

**Comment by dhitchcock** [comment]
I don't think the "same distro" matters either.  This happens with my Fedora 23 laptop and my Ubuntu 15.10 desktop.  and I hop a lot, it happened on many other combinations of fedora/ubuntu/arch.

I disabled my two party authenticator because of this reason, it's really irritating.

**Comment by PureTryOut** [comment]
I can report the same. Lately switched my laptop back to Ubuntu (but desktop on Arch), and this issue still happens. Can this sometime be fixed?

**Comment by OlliC** [comment]
Can confirm this issue. 

Hoping between Fedora Laptop and Arch Desktop daily and have to relogin everytime. 
It really starts to become an issue for me because of two-factor auth i cannot login if i forgot my phone for example.

**Comment by PureTryOut** [comment]
New laptop with Ubuntu, same issue. It's not hardware specific clearly, just Linux specific.

Seriously, can SOMEONE from Valve take a look at this and fix it? It's really annoying and it causes me to **not** setup two-factor auth.

**Comment by OlliC** [comment]
Still an issue! Can some look into this, please?

**Comment by Holzhaus** [comment]
Please make sure that you unchecked  "Don't save account credentials on this computer" in Steam -> Settings -> account. For some weird reason it was suddenly checked on my computer and caused the problem.

**Comment by nouspiro** [comment]
I have similar issue. I have dual boot Ubuntu/Windows and notebook with Ubuntu. Steam is installed on all three system. Notebook and Windows can login properly without problem. But dualboot Ubuntu can't remember login for more like 24 hours. I can restart between Windows and Ubuntu and it works. But after 24 hours I get password prompt and must enter code.

**Comment by OlliC** [comment]
@Holzhaus Yes its unchecked.

**Comment by OscarOrSomething** [comment]
Arch linux 64 bit, Steam doesn't remember any credentials.

**Comment by PhoenixtheII** [comment]
Can confirm between 2 Arch linux installations

**Comment by bradgy** [comment]
Affects me too: Ubuntu 16.04 on my XPS 13 and Ubuntu 16.04 on my HTPC.
Remember my password checked and Don't store credentials unchecked.

I have another desktop running Win7 which is not affected.

In big picture on my couch with controller especially it is a massive pain to enter my lengthy password/TFA code each time.

**Comment by ViBE-HU** [comment]
the description is not clear. for example i have Windows 10 and Ubuntu 16.04 so one Steam instance is not a Linux and i also have this issue and Steam on Ubuntu cannot save my password permanently. so i guess it's not a Linux vs. Linux or a Linux vs. Windows bug, simple a Steam Linux client bug.

**Comment by bradgy** [comment]
Interesting. Linux steam remembers my password (either on my laptop or on my HTPC) as long as I don't log into the other linux one in the meantime. If for example I only had my win7 box and my linux HTPC, I'd have to enter my password/TFA code once on each and they would be remembered from then on.

**Comment by calexil** [comment]
steam does not remember my password ever, mint 18 MATE, The default keyring is unlocked. remember password box checked, and 'do not store login cred..." unchecked... 

insanely annoying to have to enter my encrypted password over and over again. 

I think this is keyring/auto login related

**Comment by Spektakel** [comment]
tl;dr I hate reentering credentials whenever starting steam.

Same here for a couple of weeks, no problems before.

Using Ubuntu and Fedora, I change OS, use Steam once then it forgets my credentials for the other OS. I do not care much about reentering my account and password but I also have to enter my steam guard data which really sucks coz I have to look for my phone, recharge it, use this aweful guard app...

I came across a similiar problem while poking around:

If I have two steam installations for two different users on the same computer in different directories the same happens when changing the directory, eg

cd /steam/foo/ && ./steam.sh +map_background none "steam://store" -console -login foo password
cd /steam/bar/ && ./steam.sh +map_background none "steam://store" -console -login bar password

and voila, have to reenter steam guard code.

Even better this even happens when only using ONE single user with ONE operating system:

tl;dr I hate reentering credentials whenever starting steam.

cd /steam/foo/ && ./steam.sh +map_background none "steam://store" -console -login foo password
cd /steam/foo/ && ./steam.sh +map_background none "steam://store" -console -login foo password

and again, have to reenter steam guard code.

I can leave away the login part:
cd /steam/foo/ && ./steam.sh +map_background none "steam://store" -console

Now I can start steam as often as I want without entering credentials (funny, by leaving away login information I do not have to enter them over and over again). BUT as soon as I log in as another user or from another OS I have to reenter ALL credentials again, thats user, pass and guard.

(I have three screens, keyboards and mice connected to my old system and we often use the system with several people at once, even playing L4D with two people at once and a third surfing the web and yepp, it is an old but powerfull eight core Xeon system with 16GB memory and things run very smooth - two screen connected to the Geforce 8800 GTX and one to the Intel 945 GPU - but as you can see from my earlier post, the overall problem doesn't seem to be connected to this setup)

**Comment by Wolpharion** [comment]
I confirm, it's still an issue. I have Ubuntu MATE on a desktop and a laptop, and everytime I switch, I have to re-enter my credentials, even if the "Remember password" is checked.
Common, it's been three years since this post exists, and nothing has changed. What are you doing, Valve??

**Comment by storma** [comment]
Arch Desktop, Debian HTPC. This is really annoying as on my HTPC with Steam BPM + Controller, no keyboard or mouse, entering a password is a pain. Valve, you really need to get on top of this.

**Comment by Plagman** [comment]
This issue should be fixed in the next full Beta cycle. Thanks for the report.

**Comment by bradgy** [comment]
Haven't had to re-enter my password on either my home or work pcs for a few days now. Seems to be fixed for me! Thanks /u/plagman et al

**Comment by calexil** [comment]
@bradgy beta or stable?

**Comment by nouspiro** [comment]
I switched to beta on notebook. Now both desktop with normal release and notebook with beta stay logged in.

**Comment by bradgy** [comment]
i have beta on both @calexil

**Comment by OlliC** [comment]
Can confirm. I only switched one steam to beta and both stay logged in.

**Comment by darealshinji** [comment]
I have the same issue since a while, on stable and beta. I always have to re-enter my credentials. Using Ubuntu 16.04 here.

**Comment by Joinyy** [comment]
Had the issue just today on a fresh Arch install with the steam version from the official repos.

Now tried @darealshinji's suggestion and hope it will fix it.

**Comment by Suppen** [comment]
@darealshinji Any negative side effects of this? Inspecting the file with vim, it appears to contain a list of which games (by ID) are installed, which are running, and which are updating, in addition to the Steam PID when running. If Steam is not allowed to write to this file, it might create weird bugs

**Comment by darealshinji** [comment]
@Suppen Oh crap, you're right. I don't need to do this trick any longer so I had no issues. I gave it a try and it didn't automatically continue an installation after closing Steam and restarting it, but otherwise there seem to be no issues. So I guess it's used for the Steam client on start-up to figure out if it was left with something to do.

**Comment by OlliC** [comment]
I wonder why this issue is still open. Got fixed a while ago.

**Comment by kisak-valve** [comment]
Closing per the last couple comments.

**Comment by darealshinji** [comment]
Why is this closed? I still have this issue. Password is not remembered.

**Comment by kisak-valve** [comment]
Hello @darealshinji, the original issue in this issue report involving steam forgetting passwords with multiple steam installs appears to have been resolved back in October 2016. Is there something that makes your system different compared to those? What filesystem is steam installed on? Is it an encrypted home folder?

**Comment by calexil** [comment]
I can confirm that steam forgets passwords on systems with multiple accounts.

you have to reenter them when switching, every time

**Comment by darealshinji** [comment]
@kisak-valve  My system is Ubuntu 16.04 64 bits, no encrypted home folder. File system is the default one (ext4 I think). I don't know if using Steam on Linux and Windows 10 counts as "multiple Steam installs". My system has only one account but it still forgets the password when I restart Steam.

**Comment by OlliC** [comment]
So you have this problem between Windows and Linux? The original issue was only between multiple Linux installations and that got fixed.

**Comment by darealshinji** [comment]
It also happens if I only use Linux.

**Comment by Joinyy** [comment]
@darealshinji with my setup too, I am only using Linux on one pc with steam and have the issue

Am 04.08.2017 um 09:37 schrieb darealshinji <notifications@github.com<mailto:notifications@github.com>>:


It also happens if I only use Linux.

**Comment by Suppen** [comment]
Same with me. I have the problem with one computer, but not the other. They both run Kubuntu 17.04

**Comment by Suppen** [comment]
I noticed something on a place with no internet. Steam forgot me again, so I had to reenter my credentials. I mistyped my password, and Steam told me it had no connection to the Internet, and could therefore not check if the user existed. Result: I got dumped back to the login. However, when I retyped my password correctly, it went throug before giving me a dialog saying it could not connect to the Internet, and asking me if I would like to retry the connection or start in offline mode. Offline mode worked.

So it appears Steam still remembers the user and the correct credentials, but is unable to fetch them for some reason

**Comment by felipsmartins** [comment]
Same with me. From months ago it was working but does not now anymore.
Debian Jessie

**Comment by Leeo97one** [comment]
This issue is still present...
And the fact that the password isn't remembered is just one of the consequences.

It seems like a part of the Steam config is reset from time to time and the `~/.steam/registry.vdf` file seems to play an important role in this issue.

This makes the Steam experience on Linux pretty bad for a lots of people. It's a shame that there is absolulty no clues to fix it since so many years...

**Comment by romanovzky** [comment]
I have this issue as well. I have Steam on two arch linux installations and on a Windows 10 installation. Steam credentials on Arch seem to be forgotten and ignore sometimes after I play on Windows 10 (not always).

**Comment by TheNH813** [comment]
This hasn't been a issue for me except the past year or so, but it happens every time I open steam. It ALWAYS asks for the username and password for my account.

Don't worry, I have a fix. Don't do this on any computer with multiple user accounts, unless you change file permissions so only you can access your home folder. Create a shell script with the following code in it:

`#!/bin/bash
steam -login Username SteamPassword'

Obvsiously, replace Username and SteamPassword with your REAL credentials. When you open the script the steam login screen will appear, but the login button will already be pressed and the details will be filled in for you. It's that simple.

**Comment by romanovzky** [comment]
Partial workaround as it does not solve the problem for people using 2-step authentication

**Comment by Leeo97one** [comment]
It also doesn't solve the problem for people using a language other than English.

**Comment by qinq-net** [comment]
Removing writting permission for `~/.steam/registry.vdf` also works for me on Gentoo. But would it fail to boot when my session times out?

**Comment by Ishidres** [comment]
@darealshinji's method worked for me:
> I have the same issue since a while, on stable and beta. I always have to re-enter my credentials. Using Ubuntu 16.04 here.
> 
> edit:
> 
> I think I have found a solution!!
> 
> First I log in and then close Steam. Then I set the user rights of `~/.steam/registry.vdf` to read-only. Now it always logs me in automatically. By the way, that way I can change the interface language too, because it was always setting it back to English.

However, your `~/.steam/registry.vdf` **must be read and writable before you open Steam and login manually** as otherwise Steam won't be able to write to it:

1. Delete your `~/.steam/registry.vdf` and create a new empty one.
2. `chmod 755 ~/.steam/registry.vdf`
3. Open Steam, manually login and **check the "remember client credentials" option**.
4. Your `~/.steam/registry.vdf` should be written now. Next, use `chmod 0444 ~/.steam/registry.vdf` to set its permissions to read-only.
5. Close and exit Steam completely.
6. If you open Steam now, it shouldn't ask for your client credentials anymore.

Good luck! :smile::+1:

**Comment by Ishidres** [comment]
The solution I posted above doesn't seem to work on the latest version anymore, too.
Feels like valve actually tries to force Linux users to manually login on purpose... :confused:

**Comment by sikakraa** [comment]
> The solution I posted above doesn't seem to work on the latest version anymore, too.
> Feels like valve actually tries to force Linux users to manually login on purpose... confused

Steam seems to create a backup of the old file called registry.vdf.old on startup.

I copied the contents of the registry.vdf to registry.vdf.old after step 4 and the solution seemed to work after that. Seems like simple coding error and lack of tests to me.

**Comment by darealshinji** [comment]
Great, now they messed it up totally.

**Comment by Ishidres** [comment]
You can still simply prevent modification or deletion of the registry.vdf file by removing the write permission of the ~/.steam directory.  This will keep the file in the correct state to allow automatic login. However, when Steam tries to update it won't be able to write to the steam directory anymore resulting in future Steam updates to fail.

**Comment by darealshinji** [comment]
@Ishidres For me ~/.steam doesn't contain any of the actual Steam files but mostly symlinks:
```
lrwxrwxrwx 1 djcj djcj  23 Jul 18 16:26 bin -> /home/djcj/.steam/bin32
lrwxrwxrwx 1 djcj djcj  41 Jul 18 16:26 bin32 -> /home/djcj/.local/share/Steam/ubuntu12_32
lrwxrwxrwx 1 djcj djcj  41 Jul 18 16:26 bin64 -> /home/djcj/.local/share/Steam/ubuntu12_64
-r--r--r-- 1 djcj djcj 299 Jul 18 16:26 registry.vdf
-rwxrwxr-x 1 djcj djcj 299 Jul 18 16:14 registry.vdf_default
lrwxrwxrwx 1 djcj djcj  29 Jul 18 16:26 root -> /home/djcj/.local/share/Steam
lrwxrwxrwx 1 djcj djcj  37 Jul 18 16:26 sdk32 -> /home/djcj/.local/share/Steam/linux32
lrwxrwxrwx 1 djcj djcj  37 Jul 18 16:26 sdk64 -> /home/djcj/.local/share/Steam/linux64
lrwxrwxrwx 1 djcj djcj  29 Jul 18 16:26 steam -> /home/djcj/.local/share/Steam
-rw-rw-r-- 1 djcj djcj   6 Aug 10 20:20 steam.pid
prw------- 1 djcj djcj   0 Aug  7 11:32 steam.pipe
```
Updates are working fine. Not sure about steam.pid and steam.pipe but I never had issues with that.

**Comment by majamin** [comment]
@darealshinji's and @Ishidres's almost worked for me; until I logged out. My steps below are pretty much identical except that I switch TTYs between users to accomplish the changes to the permissions to `registry.vdf`. 

I have multiple accounts on my computer. For the purposes of this explanation. I am Alice, and Bob (my kid) wants to be able to open Steam without a password prompt.

1. As Alice, in a terminal `$ sudo su` to get access to Bob's home directory.
2. `$ cd /home/Bob/.steam`
3. `$ rm registry.vdf`
4. `$ touch registry.vdf && chmod 755 registry.vdf`
5. Ctrl + Alt + F2 to open tty2.
6. Log in as Bob.
7. Open Steam, using login/pass and selecting "remember password".
8. Close Steam.
9. Ctrl + Alt + F1 to open tty1 for me (Alice), and `$ chmod 0444 registry.vdf`.
10. Repeat step 5.
11. Log out of Bob's account.
12. Restart / logout and double check that everything works.

For some reason, logging out or switching user instead of the TTY switch in step 5 was what did it for me. **Please note that changing TTY may be different on your system**. Consider using `chvt` in the terminal instead of the Ctrl + Alt + F? binding.

Too soon to tell how this will break Steam updates (as may happen according to other comments).

**Comment by morkin1792** [comment]
> However, your `~/.steam/registry.vdf` **must be read and writable before you open Steam and login manually** as otherwise Steam won't be able to write to it:
> 
>     1. Delete your `~/.steam/registry.vdf` and create a new empty one.
> 
>     2. `chmod 755 ~/.steam/registry.vdf`
> 
>     3. Open Steam, manually login and **check the "remember client credentials" option**.
> 
>     4. Your `~/.steam/registry.vdf` should be written now. Next, use `chmod 0444 ~/.steam/registry.vdf` to set its permissions to read-only.
> 
>     5. Close and exit Steam completely.
> 
>     6. If you open Steam now, it shouldn't ask for your client credentials anymore.
> 
> 
> Good luck! smile+1

Currently, I use this to force the file not to be deleted:
```bash
chattr +i ~/.steam/registry.vdf
```

**Comment by Leeo97one** [comment]
It's been a while the last time I had this issue. Not sure how, but it seems to be fixed for me. :man_shrugging:

**Comment by BobbyWibowo** [comment]
Is this security related from the server-side by any chance? :thinking: 
It seems this only happens when my locations are constantly changing (e.g. switching VPN countries).
And it seems to be even worse when I have my Steam account actively logged in on a server that is located in a completely different country from my PC as well (steam cards farm bot).

I say this cause I've tried those methods above, but no luck.
I've confirmed the `registry.vdf` file hadn't been tampered with by Steam after I closed it and ran it again, cause I did both the `chmod 444` and especially `chattr +i` (it seemed like Steam would overwrite the file if I didn't do `chattr +i`).
So I concluded it was simply that the server didn't like it, instead of it being an issue with the client :thought_balloon:
